voila/configuration.py

```diff
@@ -8,11 +8,17 @@
 #############################################################################
 
 import traitlets.config
-from traitlets import Unicode, Bool, Dict, List
+from traitlets import Unicode, Bool, Dict, List, Enum
 
 
 class VoilaConfiguration(traitlets.config.Configurable):
     """Common configuration options between the server extension and the application."""
+    allow_template_override = Enum(['YES', 'NOTEBOOK', 'NO'], 'YES', help='''
+    Allow overriding the template (YES), or not (NO), or only from the notebook metadata.
+    ''').tag(config=True)
```

Comment:
Wondering whether something like `['url', 'notebook', 'all']` would sound more explicit?

Reply:
My reasoning was that you probably don't want the combination of 'permit from URL', but not from notebook metadata. Or can you come up with te use case?

Reply:
It was mostly for users who want to set the option without reading the docs or the help string, to know what 'YES' and 'NO' mean in this case.